### ISSUE TITLE
CCPA Browser / Extension level mechanism for opt-out

### ISSUE THREAD

**Issue by AramZS** [w3c/web-advertising#39]
Hello all, 

To support ongoing legal requirements around browser/extension level signals for CCPA opt-out we need a technical standard that can be commonly extended by any vendor that wishes to provide consumers with that tool. I have written a proposal to update the standard provided by the IAB to provide that interface. 

You can view the text and reasoning of the proposal along with how it would change the existing specification (as I understand it) in this pull request: https://github.com/AramZS/IAB-CCPA-Framework-Implementation-Notes/pull/2/files

**Comment by broggeri** [comment]
Hello,

This proposal requires the browser to determine that the user in in California's juridiction - is there already such a concept in the browser or would be need to define one ?

I'm interested in this as I think we should also discuss API the browser could provide to comply to European law, starting with GDPR Art. 21 which shares some similarity with CCPA 'Do Not Sell'.

**Comment by rvaneijk** [comment]
I read the proposal as is a very specific channel, based on the IAB CCPA framework. If adopted by browsers, I think it is safe to say that for GDPR we need a seperate channel. Tying Do not sell with a right to object by automated means using technical specifications may lead to confusion on what is expressed by a user. 

^@michael-oneill

**Comment by asr-enid** [comment]
In terms of the GDPR there is the much more extensive TCF 2.0 framework (from which the CCPA signalling part was derived basically) from the IAB Europe - 

- https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/tree/master/TCFv2
- https://iabeurope.eu/tcf-2-0/ 

Given the TCF not only deals with signalling an opt-out, but

- includes granular controls on the processing purposes 
- controls about involved parties (Vendors) and the status for their legal basis
- policies **HOW** this signal must be acquired UI wise (Consent Management Platforms)

the questions of how this will be integrated here is much broader (just a few that come to mind)  

- will the sandbox in general honour these APIs and signals that they provide (browser fetches a signal from the standardised API)?
- Will the browser integrate a compatible UI?
- Will the browser allow attaching a federated login/privacy management?
- how do the APIs  / processing purposes relate (under which legal basis/purposes would the browser allow FLoC based ads, Interest Group based Ads, ....)?

Stripping 3rd Party Tracking Cookies does not make these Frameworks obsolete, in that specific case it would make Purpose 1 of the TCF redundant, but with GDPR there still need to be defined legal grounds for processing of PII unrelated of 3rd party tracking.

**Comment by rodolpheAV** [comment]
> Stripping 3rd Party Tracking Cookies does not make these Frameworks obsolete, in that specific case it would make Purpose 1 of the TCF redundant, but with GDPR there still need to be defined legal grounds for processing of PII unrelated to the question of 3rd party tracking.

I'm actually trying to understand if this is true or not. 
I agree that technicaly, stripping 3rd Party Tracking Cookies is only directly connected to Purpose 1. 
But if you only manage agregated data as specified by the sandbox proposals (FLOC, Agregated reports...), you will actually not collect any PII right ? Therefore you wouldn't legaly have to define any legal ground.

**Comment by asr-enid** [comment]
`But if you only manage agregated data as specified by the sandbox proposals (FLOC, Agregated reports...), you will actually not collect any PII right ? Therefore you wouldn't legaly have to define any legal ground.`
That deserves an in-depth review once the APIs are finalised (and you know all the data flows that are actually happening), overall there is more data processing happening in the overall use-case then just with the sandbox (Prior / Post)
I can imagine that this might hold up for the FLoCs and the Aggregated Reporting itself, but there is more processing happing, for example:
1.  The advertiser that defines an interested group, does that based on profiling the user interest on his 1st-party data, which needs a legal basis. 
2. TURTLEDOVE assumes that there is an 1p-and-contextual request that injects signals into the sandbox, this request will be based on PII in most cases and needs a legal basis
...

**Comment by rodolpheAV** [comment]
@asr-enid you are right even if agregated data are not PII, depending on the data processing of each company there may be some PII usage even without third party cookie. 

Obviously geodata, or 1st party data collection by publishers would also be subject to a legal basis.

**Comment by ctavan** [comment]
> @asr-enid you are right even if agregated data are not PII, depending on the data processing of each company there may be some PII usage even without third party cookie.

It should also be noted that "personal data" in terms of GDPR goes far beyond PII. Art. 4 lit. 1 GDPR defines:

> ‘personal data’ means any information relating to an identified or identifiable natural person (‘data subject’); an identifiable natural person is one who can be identified, directly or indirectly, in particular by reference to an identifier such as a name, an identification number, location data, an online identifier or to one or more factors specific to the physical, physiological, genetic, mental, economic, cultural or social identity of that natural person;

[Recital 26](https://gdpr-info.eu/recitals/no-26/) has some more context on what is considered "anonymous data" and [recital 30](https://gdpr-info.eu/recitals/no-30/) mentions that under certain circumstances IP addresses may also be considered 'personal data'.

Willful IP Blindness may help to address the specific issue of IP addresses but I still think the observation of @asr-enid is correct in that even if the entirety of privacy sandbox features were implemented we will still very likely have some remaining types of data processing that will require TCF-like frameworks.

**Comment by AramZS** [comment]
GPC has been recognized as satisfying this requirement (and t/f satisfies my initial question as far as I'm concerned), though others are also attempting to do so. We can discuss individual efforts as separate items on the agenda so I'm going to close this ticket.
